v2/tools/generator/internal/codegen/pipeline/add_operator_spec.go

```diff
@@ -82,26 +112,171 @@ func createOperatorSpecIfNeeded(
 	result.Add(operatorSpec)
 	result.AddTypes(builder.definitions) // Add any other types that were needed as well
 
-	return result, nil
+	return result, exportedProperties, nil
+}
+
+type configMapContext struct {
+	path     []*astmodel.PropertyDefinition
+	typeName astmodel.TypeName
+}
+
+func (ctx configMapContext) withPathElement(prop *astmodel.PropertyDefinition) configMapContext {
+	return configMapContext{
+		typeName: ctx.typeName,
+		path:     append(ctx.path, prop),
+	}
+}
+
+func (ctx configMapContext) withTypeName(typeName astmodel.TypeName) configMapContext {
+	return configMapContext{
+		typeName: typeName,
+		path:     ctx.path,
+	}
+}
+
+var identityConfigMapObjectTypeVisit = astmodel.MakeIdentityVisitOfObjectType(func(ot *astmodel.ObjectType, prop *astmodel.PropertyDefinition, ctx interface{}) (interface{}, error) {
+	typedCtx := ctx.(configMapContext)
+	typedCtx = typedCtx.withPathElement(prop)
+	return typedCtx, nil
+})
+
+type configMapTypeWalker struct {
+	configuration      *config.Configuration
+	exportedProperties ExportedProperties
+	walker             *astmodel.TypeWalker
+}
+
+func newConfigMapTypeWalker(defs astmodel.TypeDefinitionSet, configuration *config.Configuration) *configMapTypeWalker {
+	result := &configMapTypeWalker{
+		configuration:      configuration,
+		exportedProperties: make(ExportedProperties),
+	}
+
+	visitor := astmodel.TypeVisitorBuilder{
+		VisitObjectType:   result.catalogObjectConfigMapProperties,
+		VisitResourceType: result.includeSpecStatus,
+	}.Build()
+	walker := astmodel.NewTypeWalker(defs, visitor)
+	walker.MakeContext = func(it astmodel.TypeName, ctx interface{}) (interface{}, error) {
+		if ctx == nil {
+			return configMapContext{
+				typeName: it,
+				path:     nil,
+			}, nil
+		}
+
+		typedCtx := ctx.(configMapContext)
+		return typedCtx.withTypeName(it), nil
+	}
+	result.walker = walker
+
+	return result
+}
+
+func (w *configMapTypeWalker) includeSpecStatus(this *astmodel.TypeVisitor, it *astmodel.ResourceType, ctx interface{}) (astmodel.Type, error) {
+	typedCtx := ctx.(configMapContext)
+
+	// Fabricate a fake PropertyDetails here
+	prop := astmodel.NewPropertyDefinition("Status", "status", it.StatusType()) // TODO: This isn't actually guaranteed right now?
```

Comment:
Do you need to take care of this `TODO` before merging?

Reply:
Actually, call `it.Property("Status")` and you'll get the property you need. `ResourceType` implements `PropertyContainer`.

Reply:
That's part of the problem, but the other part of the problem is that there's no guarantee elsewhere in the code that the path actually is part of status. So probably I need to rewrite this to append spec to the spec path and status to the status path.

Basically the issue here is I am making the assumption that we're going to take the `Status` branch, but that's not actually required (although in practice it is because we won't export _spec_ fields it'd be weird)

Reply:
Fix this.